### PR TITLE
feat(perf): change detection profiler

### LINKED ISSUE #3877: Implement programmatic change detection profiler
The profiler would be accessible via dev tools. When a developer installs the profiler DI module they get access to global functions they could invoke directly from the dev tools console.

The profiler would be exposed under global namespace `ng` (which in the future could host more tools). To profile change detection specifically, a developer would enter:

```javascript
ng.profileChangeDetection();
```

The profiler would invoke `LifeCycle.tick()` at least 5 times and for at least 500ms. It will then compute the average and the standard deviation of amount of time it took to execute a single `tick`.

`profileChangeDetection` takes an optional named boolean parameter `record` that's `false` by default. When set to `true`, it records the profile as a CPU profile that could be inspected later in the "Profiles" tab of the dev tools:

```javascript
ng.profileChangeDetection({ record: true });
```

In the future, we may add more options as named parameters, such as custom number of `tick` runs or custom length of profile.

/cc @vsavkin, @mhevery

### PR DESCRIPTION
**Pull request by yjbanov**
Fixes #3877

### CODE REVIEW

File: modules/angular2/test/toolz/spies.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'package:angular2/test_lib.dart' show SpyObject;
```
Comment (vsavkin) [inline modules/angular2/test/toolz/spies.dart]: why toolz and not tools?

File: modules/angular2/test/toolz/spies.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {SpyObject} from 'angular2/test_lib';
+import {ApplicationRef, LifeCycle, Injector, bind} from 'angular2/angular2';
+import {global} from 'angular2/src/core/facade/lang';
+
+export class SpyApplicationRef extends SpyObject {
+  injector;
+  constructor() {
+    super();
+    var injector = Injector.resolveAndCreate([
```
Comment (vsavkin) [inline modules/angular2/test/toolz/spies.ts]: this.injector = Injector.resolveAndCreate

Comment (vsavkin) [inline modules/angular2/test/toolz/spies.ts]: Or you can rewrite the whole thing as follows:

```
class SpyApplicationRef {
  injector = Injector.resolveAndCreate([...]);
}
```

File: modules/angular2/test/toolz/spies.dart (hunk 1 above)
Comment (yjbanov) [inline modules/angular2/test/toolz/spies.dart]: There's something in our build pipeline that doesn't like directories named `tools`. It causes Karma to look for `_spec` files _outside_ `modules`. I couldn't figure it out, so for now making it `toolz`.

Comment (yjbanov) [inline modules/angular2/test/toolz/spies.dart]: Igor helped me fix this (some weird special case in Karma config)

File: modules/angular2/test/toolz/spies.ts (hunk 2 above)
Comment (yjbanov) [inline modules/angular2/test/toolz/spies.ts]: Done

**Pull request closed without merging**